src/status.rs

```diff
@@ -109,11 +148,124 @@ pub fn show_build_status(
 				}
 			}
 		}
-		println!("Building for {}...", MinSec::since(start_time));
-		if workers.iter().all(|worker| *worker == WorkerStatus::Done) {
+
+		let build_is_done = buildstate.all_workers_done();
+
+		// Compute remaining time for this build, by simulating a build.
+		let mut simulated_time = Instant::now();
+		let mut estimation_impossible = false;
+		while !buildstate.all_workers_done_or_idle() {
+			// Give all simulated workers something to do
+			for i in 0..buildstate.workers.len() {
+				let worker = &buildstate.workers[i];
+				match worker {
+					WorkerStatus::Starting | WorkerStatus::Idle => {
+						let next = queuestate.next_at(simulated_time);
+						buildstate.set_status(
+							i,
+							match next {
+								Some(task) => WorkerStatus::Running { task },
+								None => WorkerStatus::Idle,
+							},
+						);
+					}
+					_ => {}
+				}
+			}
+
+			// Find the job with the lowest remaining time
+			let (worker, task, remainingtime) = match buildstate
+				.workers
+				.iter()
+				.enumerate()
+				.flat_map(|(i, worker)| match worker {
+					WorkerStatus::Running { task, .. } => match queuestate.get_task_status(*task) {
+						TaskStatus::Running { start_time, .. } => {
+							let runtime = if simulated_time > start_time {
```

Comment:
Note: If you want, you could make your own extension trait to add `saturating_sub` and/or `checked_sub` to `Instant`.

Reply:
Example: https://www.notion.so/Extension-Trait-Example-5a5fe26c9388475ab1c78db4a413e3a3